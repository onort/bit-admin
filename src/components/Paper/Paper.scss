@import "../../styles/main.scss";

.container {
  width: 100%;
  background-color: $white;
  padding: $spacer-xlg;
  border-radius: px-to-rem(2px);
}

.elevation1 {
  box-shadow: 0 2px 1px -1px rgba(0, 0, 0, 0.2), 0 1px 1px 0 rgba(0, 0, 0, 0.14),
    0 1px 3px 0 rgba(0, 0, 0, 0.12);
}

.elevation2 {
  box-shadow: 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 2px 2px 0 rgba(0, 0, 0, 0.14),
    0 1px 5px 0 rgba(0, 0, 0, 0.12);
}

.elevation3 {
  box-shadow: 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 3px 4px 0 rgba(0, 0, 0, 0.14),
    0 1px 8px 0 rgba(0, 0, 0, 0.12);
}

.elevation4 {
  box-shadow: 0 2px 4px -1px rgba(0, 0, 0, 0.2), 0 4px 5px 0 rgba(0, 0, 0, 0.14),
    0 1px 10px 0 rgba(0, 0, 0, 0.12);
}
