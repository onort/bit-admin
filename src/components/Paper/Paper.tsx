import React from "react"
import cx from "classnames"

import styles from "./Paper.scss"

type elevationType = 1 | 2 | 3 | 4

interface Props {
  className?: string
  elevation: elevationType
}

const Paper: React.SFC<Props> = props => {
  const className = cx(
    styles.container,
    { [styles[`elevation${props.elevation}`]]: props.elevation },
    props.className
  )
  return <div className={className}>{props.children}</div>
}

export default Paper
