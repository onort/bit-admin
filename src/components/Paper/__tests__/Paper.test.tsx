import React from "react"
import { shallow } from "enzyme"

import Paper from "../"

describe("<Paper />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<Paper elevation={1} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have correct elevation class", () => {
    const wrapper = shallow(<Paper elevation={1} />)
    expect(wrapper.find("div").hasClass("elevation1")).toBe(true)
  })

  it("should be able to have a custom className", () => {
    const testClassName = "test"
    const wrapper = shallow(<Paper elevation={1} className={testClassName} />)
    expect(wrapper.find("div").hasClass(testClassName)).toBe(true)
  })
})
