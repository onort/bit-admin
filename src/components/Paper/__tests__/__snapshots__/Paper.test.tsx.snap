// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paper /> should match snapshot 1`] = `
<div
  className="container elevation1"
/>
`;

exports[`<Paper /> should match snapshot 2`] = `"<div class=\\"container elevation1\\"></div>"`;
